spec/function.md

```diff
@@ -9,83 +9,67 @@ In other words, every function can only receive argument with type of record and
 
 All function application in Topoi must follow the following syntax rule:
 
+### Input with 0 field
+Say:
 ```
-A . F
+f : () -> R
 ```
-or
+then invocation of `f` will be:
 ```
-A . F E
+. f
 ```
-Note that function application is left-associative (like number addition):
+and:
 ```
-A . F E . F E = (A . F E) . F E
-
-similar to
-
-1 + 2 + 3 = (1 + 2) + 3
-```
-
-Where 
-```
-A: (any type)
-F: Function
-E: Record
+(. f) : R
 ```
+Note that this syntax is not associative, `..f` does not mean `.(.f)`, it is simply invalid.
 
-Suppose `F`'s input type is `Record` with 1 field:
+### Input with 1 field
+Say:
 ```
-F: (K1: T1) -> R
+f: (k: T) -> R
 ```
-And:
-```
-A: T1
-```
-Then:
+and
 ```
-(A . F) : R
+e: T
 ```
-
-
-
-Suppose `F`'s input type is `Record` with more than 1 field:
+then the invocation of `f` will be:
 ```
-F: (K1: T1, K2: T2, K3: T3) -> R
+e . f
 ```
 and
 ```
-A: T1
-```
-then
+(e . f) : R
 ```
-E: (K2: T2, K3: T3)
+Note that this syntax is left-associative, meaning that:
 ```
-and
+e . f . g = (e . f) . g
 ```
-(A . F E) : R
-```
-Similarly, if 
+
+### Input with more than 1 fields
+Say:
 ```
-A: T2
+f : (j: T, k: U, l: V) -> R
 ```
-then
+and:
 ```
-E: (K1: T1, K3: T3)
+a: T
+b: U
+c: V
 ```
-and
+then the invocation of `f` can be any of the following:
 ```
-(A . F E) : R
+a.f(k:=b, l:=c)
+b.f(j:=a, l:=c)
```

Comment:
How can we in-place destruct the value by the key here? @wongjiahau

Reply:
What do you mean ? Can you show me an example of what you intend to achieve?

Reply:
Say, I got the function that accepts `f : (x: A, y: B) -> ...`.
And, I got a function `someFunc : ... -> (z: B)`.

How can I marshal the input for function `f` in the chain function? eg.

`foo.someFunc.f(x:= a, y:=b)`

is this correct? @wongjiahau